### ISSUE TITLE
SMSG_UPDATE_OBJECT unk fields

### ISSUE THREAD

**Issue by Artamedes** [ProjectSkyfire/SkyFire_548#837]
**Description:**

SMSG_UPDATE_OBJECT movement part has a lot of unk fields
this is what I got from my research so far, number,offset
`0. 0    UPDATEFLAG_NO_BIRTH_ANIM
1. 1    UPDATEFLAG_ENABLE_PORTALS
3. 2    UPDATEFLAG_PLAY_HOVER_ANIM
4. 3    UPDATEFLAG_SUPPRESSED_GREETINGS
5. 368  UPDATEFLAG_HAS_MOVEMENT_UPDATE
6. 424  UPDATEFLAG_HAS_TRANSPORT_POSITION
7. 448  UPDATEFLAG_HAS_POSITION
8. 464  UPDATEFLAG_HAS_COMBAT_VICTIM
9. 476  UPDATEFLAG_HAS_SERVER_TIME
10.488  UPDATEFLAG_HAS_VEHICLE_CREATE
11.498  UPDATEFLAG_HAS_ANIMKITS_CREATE
12.512  UPDATEFLAG_HAS_ROTATION
13.668  UPDATEFLAG_HAS_AREATRIGGER
14.676  UPDATEFLAG_HAS_GAMEOBJECT
15.680  UPDATEFLAG_THIS_IS_YOU
16.681  UPDATEFLAG_REPLACE_YOU
17.810  UPDATEFLAG_SCENE_LOCAL_SCRIPT_DATA
17.1032 UPDATEFLAG_SCENE_OBJECT
18.1044 
19.1064 UPDATEFLAG_SCENE_PENDING_INSTANCES
20.1068 UPDATEFLAG_TRANSPORT_FRAME_COUNT`

here's also my movement update code
https://gist.github.com/Artamedes/57033cf67f5c538860bf90ae5b8fe9cc#file-objectupdate-cpp-L39 ( i have pretty much all done except 1044 and scene object

**SF rev. hash/commit:** 3bf376f0a1bb69225a23d57b166b32963daa287e

**Comment by SkyFire** [comment]
A pull request would make it easier to review your code up against existing code. Even if changes are needed it it can be easier do modify.

**Comment by Artamedes** [comment]
unfortunately my code is more than likely not going to work with the core, since some implementations would be needed, 1 being boost for optional. should i make a pull with tons of these implementations?

**Comment by Artamedes** [comment]
also i'm thinking 1044 is lowguid

**Comment by AriDEV** [comment]
SceneObjCreate contains LocalScriptData and PetBattleFullUpdate

**Comment by Artamedes** [comment]
will reverse sceneobject soon and probably make a pull, but pull will include boost (any objection?)

**Comment by Artamedes** [comment]
https://gist.github.com/Artamedes/57033cf67f5c538860bf90ae5b8fe9cc/revisions

gist has been updated, there was some stuff wrong with it if you want to take a look @AriDEV

**Comment by AriDEV** [comment]
are you sure that the faceguid mask is send after all other bytes else your position is wrong :/ i guess it should be before sending the other bytes ?

**Comment by Artamedes** [comment]
that's order client reads it in the movementupdate block, it reads at end for some reason

https://github.com/ProjectSkyfire/SkyFire_548/blob/master/src/server/game/Entities/Object/Object.cpp#L729

https://github.com/ProjectSkyfire/SkyFire_548/blob/master/src/server/game/Movement/Spline/MovementPacketBuilder.cpp#L351

already done here

**Comment by Artamedes** [comment]
@AriDEV https://gist.github.com/Artamedes/f19691a7896b2cd6af92a8758621ca92#file-readmovementupdateblock-L1188

**Comment by AriDEV** [comment]
@Artamedes yes this looks for me like facing target part its allready on https://github.com/ProjectSkyfire/SkyFire_548/blob/master/src/server/game/Entities/Object/Object.cpp#L728

**Comment by SkyFire** [comment]
Old Issue. Fixed long time ago
